### PR TITLE
Convert specs to RSpec 3.3.1 syntax with Transpec

### PR DESCRIPTION
**Pull request by jhawthorn**
Just let transpec do it's thing on all projects.

```bash
for project in api core frontend backend sample; do
  transpec -f $project/spec -c 'bundle exec rake test_app && bundle exec rspec'
  git commit -aF .git/COMMIT_EDITMSG
done
```

transpec is **so** good.

### CODE REVIEW

File: api/spec/support/controller_hacks.rb [hunk 1]
```diff
@@ -31,4 +31,15 @@ def api_process(action, params={}, session=nil, flash=nil, method="get")
...
+  # rspec-rails 3 will no longer automatically infer an example group's spec type
+  # from the file location. You can explicitly opt-in to the feature using this
+  # config option.
+  # To explicitly tag specs without using automatic inference, set the `:type`
+  # metadata manually:
+  #
+  #     describe ThingsController, :type => :controller do
+  #       # Equivalent to being in spec/controllers
+  #     end
+  config.infer_spec_type_from_file_location!
```
Comment (adammathys) [inline api/spec/support/controller_hacks.rb]: We probably don't need this.

**Comment by adammathys** [comment]
:+1: Outside of those couple of config things.

**Comment by jhawthorn** [comment]
updated

**Comment by jordan-brough** [comment]
:+1:

**Comment by magnusvk** [comment]
I like this new syntax so much better... Thanks for doing this.

**Pull request merged**